docs/content/guides/dagster/local-to-production-tutorial.mdx

```diff
@@ -0,0 +1,420 @@
+def repo():
+    resource_defs = {
+        "hn_client": hn_api_client,
+        "io_manager": snowflake_io_manager.configured(
+            {
+                "account": {"env": "SNOWFLAKE_ACCOUNT"},
+                "user": {"env": "SNOWFLAKE_USER"},
+                "password": {"env": "SNOWFLAKE_PASSWORD"},
+                "database": {"env": "SNOWFLAKE_DATABASE"},
+                "schema": {"env": "SNOWFLAKE_SCHEMA"},
+            }
+        ),
+    }
+
+    return [*with_resources([items, comments, stories], resource_defs=resource_defs)]
+```
+
+<Note>
+
+Since all of the Snowflake IO manager configuration is being set through environment variables, we no longer need to select the resource configuration based on deployment! If we decide to add more configuration that varies between local and production, we can go back to the previous pattern of selecting config based on the deployment name.
+
+</Note>
+
+#### Staging
+
+Depending on your organization’s Dagster setup, there are a couple of options for a staging environment.
+
+If you are a Dagster Cloud user, we recommend using [Branch Deployments](https://docs.dagster.cloud/guides/branch-deployments) as your staging step. Check out our \[comprehensive guide to Branch Deployments]\(link to claire’s doc) to learn how to use Branch Deployments to verify your data pipelines before deploying them to production.
+
+If you host a staging deployment of Dagster, we’ve already done all of the necessary work to run our assets in staging! Since we get the configuration for our assets from environment variables, we can simply set the variables to new values in the staging environment, and we’re all set.
+
+### Conclusion
+
+This guide demonstrates how we recommend that you write your assets and jobs so that they transition from local development to staging and production environments without requiring code changes at each step. While we focused on assets in this guide, the same concepts and APIs can be used to swap out run configuration for jobs.
+
+### Optional: Testing
+
+You may have noticed a missing step in the development workflow presented in this guide— unit tests! While the main purpose of the guide is to help you transition your code from local development to a production deployment, unit testing is still an important part of the development cycle, and we’d like to take this time to explore a pattern you may find useful when writing your own unit tests: stubbing resources.
+
+The general idea behind stubbing or mocking a Dagster resource is relatively simple: since Dagster’s resource system allows you to supply different implementations of a resource to an asset or op, we can make a simplified version of a “real” resource and use it in unit tests!
+
+<Note>
+  Determining when it makes sense to stub a resource for a unit test can be a
+  topic of much debate. There are certainly some resources where it would be too
+  complicated to write and maintain a stub. In general, if a resource is
```

Comment:
If you have an example here of such a resource, it would be helpful. Fine if not though.

Reply:
yeah adding an example would be great. Some examples that come to mind are Snowflake, aws/gcp/azure